src/pages/Swap/index.tsx

```diff
@@ -118,6 +127,29 @@ export default function Swap({ history }: RouteComponentProps) {
     inputError: swapInputError,
   } = useDerivedSwapInfo(toggledVersion)
 
+  const toggleVersionCallback = useToggleVersionCallback()
+
+  // automatically toggle best version
+  useEffect(() => {
+    if (!bestTrade || !trade) return
+
+    if (bestTrade !== trade) {
+      toggleVersionCallback()
```

Comment:
i would be more comfortable with this explicitly changing to the bestTrade version, but also i think we don't need to switch via the url, just always use the best trade right?

Reply:
agreed. changing it so that url param forces a version, no url param just uses the best trade without redirecting.